crates/typst/src/text/font/book.rs

```diff
@@ -93,8 +93,8 @@ impl FontBook {
         variant: FontVariant,
         text: &str,
     ) -> Option<usize> {
-        // Find the fonts that contain the text's first char ...
-        let c = text.chars().next()?;
+        // Find the fonts that contain the text's first non-space char ...
+        let c = text.chars().find(|c| !c.is_whitespace())?;
```

Comment:
What happens if all characters are whitespace? I believe that'd receive the same treatment as an empty string, which shouldn't be that bad, right?

Reply:
when the whole string is space, it wont even be shaped. there is a `if` handling this

Reply:
Got it, thanks!